segfault in Document.Parse htmlMutex

Caught this panic on our Ubuntu Linux build server using latest (0.4.1) binary release:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x50e756]

goroutine 1 [running]:
github.com/wjdp/htmltest/htmldoc.(*Document).Parse(0x0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmldoc/document.go:47 +0x26
github.com/wjdp/htmltest/htmldoc.(*Document).IsHashValid(0x0, 0xc427267419, 0x22, 0xc4200b7800)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmldoc/document.go:105 +0x2b
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkInternalHash(0xc4200b7680, 0xc4272c07b0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/check-link.go:284 +0xbb
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkInternal(0xc4200b7680, 0xc4272c07b0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/check-link.go:258 +0xe9
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkLink(0xc4200b7680, 0xc4203becb0, 0xc42727c5b0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/check-link.go:84 +0x3d4
github.com/wjdp/htmltest/htmltest.(*HTMLTest).testDocument(0xc4200b7680, 0xc4203becb0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/htmltest.go:146 +0x165
github.com/wjdp/htmltest/htmltest.(*HTMLTest).testDocuments(0xc4200b7680)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/htmltest.go:130 +0x69
github.com/wjdp/htmltest/htmltest.Test(0xc420019fb0, 0x1)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/htmltest.go:93 +0x771
main.run(0xc420019fb0, 0x7ffd07310b5e)
	/home/travis/gopath/src/github.com/wjdp/htmltest/main.go:154 +0x29e
main.main()
	/home/travis/gopath/src/github.com/wjdp/htmltest/main.go:62 +0x1c1
Process exited with code 2
```
I don't know the cause yet, but it's random and it was actually first time I saw it.

I get this output about once per 10 runs, today I've got slightly different output indicating another line number:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x50f986]

goroutine 1 [running]:
github.com/wjdp/htmltest/htmldoc.(*Document).Parse(0x0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmldoc/document.go:46 +0x26
github.com/wjdp/htmltest/htmldoc.(*Document).IsHashValid(0x0, 0xc4292f58af, 0xb, 0xc4200b7800)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmldoc/document.go:106 +0x2b
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkInternalHash(0xc4200b7680, 0xc426057800)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/check-link.go:292 +0xbb
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkInternal(0xc4200b7680, 0xc426057800)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/check-link.go:266 +0xe9
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkLink(0xc4200b7680, 0xc420493c70, 0xc4291c02a0)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/check-link.go:91 +0x419
github.com/wjdp/htmltest/htmltest.(*HTMLTest).testDocument(0xc4200b7680, 0xc420493c70)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/htmltest.go:197 +0x165
github.com/wjdp/htmltest/htmltest.(*HTMLTest).testDocuments(0xc4200b7680)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/htmltest.go:181 +0x69
github.com/wjdp/htmltest/htmltest.Test(0xc420019fb0, 0x1, 0x1, 0x49)
	/home/travis/gopath/src/github.com/wjdp/htmltest/htmltest/htmltest.go:141 +0xb2e
main.run(0xc420019fb0, 0x7ffd3c800b5e)
	/home/travis/gopath/src/github.com/wjdp/htmltest/main.go:155 +0x29e
main.main()
	/home/travis/gopath/src/github.com/wjdp/htmltest/main.go:62 +0x1c1
Process exited with code 2
```

--- a/htmltest/check-link.go
+++ b/htmltest/check-link.go
@@ -254,14 +254,15 @@ func (hT *HTMLTest) checkInternal(ref *htmldoc.Reference) {
 				Message:   "target is a directory, href lacks trailing slash",
 				Reference: ref,
 			})
+			refExists = false
 		}
 	} else {
 		// If that fails attempt to lookup with filesystem, resolve a path and check
 		refOsPath := path.Join(hT.opts.DirectoryPath, ref.RefSitePath())
-		hT.checkFile(ref, refOsPath)
+		refExists = hT.checkFile(ref, refOsPath)
 	}
 
-	if len(ref.URL.Fragment) > 0 {
+	if refExists && len(ref.URL.Fragment) > 0 {
 		// Is also a hash link
 		hT.checkInternalHash(ref)
 	}
@@ -308,7 +309,7 @@ func (hT *HTMLTest) checkInternalHash(ref *htmldoc.Reference) {
 	}
 }
 
-func (hT *HTMLTest) checkFile(ref *htmldoc.Reference, absPath string) {
+func (hT *HTMLTest) checkFile(ref *htmldoc.Reference, absPath string) bool {
 	f, err := os.Stat(absPath)
 	if os.IsNotExist(err) {
 		hT.issueStore.AddIssue(issues.Issue{
@@ -316,7 +317,7 @@ func (hT *HTMLTest) checkFile(ref *htmldoc.Reference, absPath string) {
 			Message:   "target does not exist",
 			Reference: ref,
 		})
-		return
+		return false
 	}
 	output.CheckErrorPanic(err)
 
@@ -326,7 +327,9 @@ func (hT *HTMLTest) checkFile(ref *htmldoc.Reference, absPath string) {
 			Message:   "target is a directory, no index",
 			Reference: ref,
 		})
+		return false
 	}
+	return true
 }
 
 func (hT *HTMLTest) checkMailto(ref *htmldoc.Reference) {

